src/node.rs

```diff
@@ -301,33 +331,409 @@ impl Taffy {
+    let node = taffy.new_with_children(FlexboxLayout::default(), &[child0, child1]).unwrap();
+
+    // Nodes should be present
+    assert!(taffy.find_node(node).is_ok());
+    assert!(taffy.find_node(child0).is_ok());
+    assert!(taffy.find_node(child1).is_ok());
+
+    taffy.clear();
+
+    // nodes should not be present
+    assert!(taffy.find_node(node).is_err());
+    assert!(taffy.find_node(child0).is_err());
+    assert!(taffy.find_node(child1).is_err());
+}
+
+// #[test]
+// fn clear_should_invalidate_id() {
+//     // I'm not sure how to test this
+//     todo!("Test is missing");
+// }
+
+#[test]
+fn remove_node_should_remove() {
+    let mut taffy = Taffy::new();
+
+    let node = taffy.new_with_children(FlexboxLayout::default(), &[]).unwrap();
+
+    // node should exist
+    assert!(taffy.find_node(node).is_ok());
+
+    let _ = taffy.remove(node).unwrap();
+
+    // node should no longer exist
+    assert!(taffy.find_node(node).is_err());
+}
+
+#[test]
+fn remove_node_should_detach_herarchy() {
+    let mut taffy = Taffy::new();
+
+    // Build a linear tree layout: <0> <- <1> <- <2>
+    let node2 = taffy.new_with_children(FlexboxLayout::default(), &[]).unwrap();
+    let node1 = taffy.new_with_children(FlexboxLayout::default(), &[node2]).unwrap();
+    let node0 = taffy.new_with_children(FlexboxLayout::default(), &[node1]).unwrap();
+
+    // Node1 should exist
+    assert!(taffy.find_node(node1).is_ok());
+
+    // Both node0 and node1 should have 1 child nodes
+    assert_eq!(taffy.children(node0).unwrap().as_slice(), &[node1]);
+    assert_eq!(taffy.children(node1).unwrap().as_slice(), &[node2]);
+
+    // Disconnect the tree: <0> <2>
+    let _ = taffy.remove(node1).unwrap();
+
+    // Node1 should no longer exist
+    assert!(taffy.find_node(node1).is_err());
+
+    // Both remaining nodes should have no child nodes
+    assert!(taffy.children(node0).unwrap().is_empty());
+    assert!(taffy.children(node2).unwrap().is_empty());
+}
+
+#[test]
+fn remove_last_node() {
+    let mut taffy = Taffy::new();
+
+    let parent = taffy.new_with_children(FlexboxLayout::default(), &[]).unwrap();
+    let child = taffy.new_with_children(FlexboxLayout::default(), &[]).unwrap();
+    taffy.add_child(parent, child).unwrap();
+
+    taffy.remove(child).unwrap();
+    taffy.remove(parent).unwrap();
+}
+
+#[test]
+fn set_measure() {
+    let mut taffy = Taffy::new();
+    let node =
+        taffy.new_leaf(FlexboxLayout::default(), MeasureFunc::Raw(|_| Size { width: 200.0, height: 200.0 })).unwrap();
+    taffy.compute_layout(node, Size::undefined()).unwrap();
+    assert_eq!(taffy.layout(node).unwrap().size.width, 200.0);
+
+    taffy.set_measure(node, Some(MeasureFunc::Raw(|_| Size { width: 100.0, height: 100.0 }))).unwrap();
+    taffy.compute_layout(node, Size::undefined()).unwrap();
+    assert_eq!(taffy.layout(node).unwrap().size.width, 100.0);
+}
+
+#[test]
+/// Test that adding `add_child()` works
```

Comment:
I _think_ it's more idiomatic to have the comment above the `#[test]`, but I'm not 100% sure on that

Reply:
Done